### DISCUSSION TITLE
Hide/Show Widget programmatically

### DISCUSSION THREAD

**Discussion by KrishKash** [iTwin/itwinjs-core#7016] [Q&A]
Hi there,
I want to hide or show widgets based on selection of grid data and/or switching widgets.
Here is a sandbox for same- [Hide-or-Show-Widgets](https://www.itwinjs.org/sandboxes/Krishna%20Kashyap/Hide-or-Show-Widgets)

Scenario- 
1. When any row is clicked in Widget1:  Widget2 & Widget3 should be visible in the top left panel
2. When Widget2 is visible:  Bottom2-1 & Bottom2-2 widgets should be visible in the bottom left panel
3. When Widget3 is clicked: Bottom3-1 & Bottom3-2 should be visible and other widgets in bottom panel should get hidden

Could someone please guide on how to achieve this?
Thanks

**Comment by grigasp** [comment]
@GerardasB

**Reply by KrishKash** [reply]
Hey @GerardasB 
Any idea on what configuration I might be missing here?

**Comment by GerardasB** [comment]
Hey @KrishKash,
Use [UiFramework.frontstages.activeFrontstagedef](https://www.itwinjs.org/reference/appui-react/frontstage/frameworkfrontstages/#activefrontstagedef) to get the frontstage in which you can [find your widgets](https://www.itwinjs.org/reference/appui-react/frontstage/frontstagedef/#findwidgetdef) by their id. Then you'll want to call [setWidgetState](https://www.itwinjs.org/reference/appui-react/widget/widgetdef/#setwidgetstate).

Here's a code sample: https://github.com/iTwin/appui/blob/7f2bd525dfa9d4e4cebd1030f5166d61eceab4de/test-apps/appui-test-app/appui-test-providers/src/ui/widgets/LayoutWidget.tsx#L499
